heat/core/io.py

```diff
@@ -354,36 +395,181 @@ def save_netcdf(data, path, variable, mode="w", **kwargs):
+            """ Using var[var_slices][data_slices] = data
+            combines a __getitem__ with a __setitem__ call, therefore it does not allow
+            parallelization of the write-operation and does not work is var_slices = slice(None)
+            (in that casem __getitem__ returns a copy and not a view).
+            This method merges both keys:
+            var[mergeSlices(var, var_slices, data)] = data
+
+            Parameters
+            ----------
+            var : netcdf4.Variable
+                Variable to which data is to be saved.
+            var_slices :
+                Keys to pass to the set-operator.
+            data : dndarray
+                Data to be saved.
+            data_slices: tuple of slices
+                As returned by the data.comm.chunk method.
+
+            Returns
+            -------
+
+                Key for the set-operation.
+
+            Raises
+            -------
+
+            """
+            slices = data_slices
+            if slices is None:
+                _, _, slices = data.comm.chunk(data.gshape, data.split if is_split else 0)
+            start, count, stride, _ = nc.utils._StartCountStride(
+                elem=var_slices,
+                shape=var.shape,
+                dimensions=var.dimensions,
+                grp=var.group(),
+                datashape=data.shape,
+                put=True,
+            )
+            out_shape = nc._netCDF4._out_array_shape(count)
+            out_split = getExpandedSplit(data.shape, out_shape, data.split)
+
+            start, count, stride = start.T, count.T, stride.T  # transpose for iteration
+            stop = start + stride * count
+            new_slices = []
+            for begin, end, step in zip(start, stop, stride):
+                if begin.size == 1:
+                    begin, end, step = begin.item(), end.item(), step.item()
+                    new_slices.append(slice(begin, end, step))
+                else:
+                    begin, end, step = begin.flatten(), end.flatten(), step.flatten()
+                    new_slices.append(
+                        np.r_[
+                            tuple(
+                                slice(b.item(), e.item(), s.item())
+                                for b, e, s in zip(begin, end, step)
+                            )
+                        ]
+                    )
+            if out_split is not None:  # add split-slice
+                if isinstance(new_slices[out_split], slice):
+                    start, stop, step = (
+                        new_slices[out_split].start,
+                        new_slices[out_split].stop,
+                        new_slices[out_split].step,
+                    )
+                    sliced = range(start, stop, step)[slices[data.split]]
+                    a, b, c = sliced.start, sliced.stop, sliced.step
+                    a = None if a < 0 else a
+                    b = None if b < 0 else b
```

Comment:
Actually, it might be the most elegant solution to just use `sliced` (which is of type range) as index instead of transforming it into a slice

Reply:
> Actually, it might be the most elegant solution to just use sliced (which is of type range) as index instead of transforming it into a slice

No, even though slicing with ranges is generally possible, this fails the `indexing netcdf file: broadcasting var`-test.

I don't think step needs to be corrected because it is generated by nc.utils (so possible 'weird' values are already dealt with) and is used to calculate stop. Also the slicing doesn't change the step because ht.comm.chunk() always uses `None` as step.